### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by bmorel**
adds some comments to try to make things easier to get in mind, address minor refactoring things, silence a warning while at it.

### CODE REVIEW

File: src/cgame/cg_main.cpp [hunk 1]
```diff
@@ -1119,7 +1119,6 @@ void CG_Init( int serverMessageNum, int clientNum, const glconfig_t& gl, const G
 	new(&cgs) cgs_t{}; // Using {} instead of () to work around MSVC bug
 	cg.~cg_t();
 	new(&cg) cg_t{};
-	memset( cg_entities, 0, sizeof( cg_entities ) );
```
Comment (slipher) [inline src/cgame/cg_main.cpp:1122]: This is just deleting the initialization with no replacement. You could do `for(centity_t& ent : cg_entities) { ent = {}; }`

**Comment by bmorel** [comment]
comments fixed

File: src/shared/bg_pmove.cpp [hunk 2]
```diff
@@ -2963,15 +2963,7 @@ static void PM_GroundTrace()
...
-				if ( pm->ps->stats[ STAT_STATE ] & SS_WALLCLIMBING )
-				{
-					pm->ps->stats[ STAT_STATE ] &= ~SS_WALLCLIMBING;
-				}
-				else
-				{
-					pm->ps->stats[ STAT_STATE ] |= SS_WALLCLIMBING;
-				}
-
+				pm->ps->stats[ STAT_STATE ] ^= ~SS_WALLCLIMBING;
```
Comment (slipher) [inline src/shared/bg_pmove.cpp:2966]: This toggles every flag except wallclimbing

Comment (bmorel) [reply src/shared/bg_pmove.cpp]: The joy of cut & paste carelessly... sorry, and fixed.

**Comment by slipher** [comment]
LGTM

**Pull request merged**